src/http/client.cr

```diff
@@ -380,94 +380,96 @@ class HTTP::Client
     end
   {% end %}
 
-  # Executes a POST with form data. The "Content-type" header is set
-  # to "application/x-www-form-urlencoded".
-  #
-  # ```
-  # client = HTTP::Client.new "www.example.com"
-  # response = client.post_form "/", "foo=bar"
-  # ```
-  def post_form(path, form : String | IO, headers : HTTP::Headers? = nil) : HTTP::Client::Response
-    request = new_request("POST", path, headers, form)
-    request.headers["Content-type"] = "application/x-www-form-urlencoded"
-    exec request
-  end
+  {% for http_method in %w(post put patch) %}
+    # Executes a {{http_method.id.upcase}} with form data. The "Content-type" header is set
+    # to "application/x-www-form-urlencoded".
+    #
+    # ```
+    # client = HTTP::Client.new "www.example.com"
+    # response = client.{{http_method.id}}_form "/", "foo=bar"
+    # ```
+    def {{http_method.id}}_form(path, form : String | IO, headers : HTTP::Headers? = nil) : HTTP::Client::Response
+      request = new_request({{http_method.upcase}}, path, headers, form)
+      request.headers["Content-type"] = "application/x-www-form-urlencoded"
```

Comment:
While you're at it, could you please change all `Content-type` to `Content-Type` with capital `T`? Apart from this file we have it this way everywhere and that's the spelling recommended by RFC 7231.